jerry-core/api/jerry.c

```diff
@@ -3078,6 +3086,57 @@ jerry_resolve_or_reject_promise (jerry_value_t promise, /**< the promise value *
 #endif /* ENABLED (JERRY_ES2015_BUILTIN_PROMISE) */
 } /* jerry_resolve_or_reject_promise */
 
+/**
+ * Get the result of a promise.
+ *
+ * @return - Promise result
+ *         - Type error if the promise support was not enabled or the input was not a promise object
+ */
+jerry_value_t
+jerry_get_promise_result (const jerry_value_t promise) /**< promise object to get the result from */
+{
+#if ENABLED (JERRY_ES2015_BUILTIN_PROMISE)
+  if (!jerry_value_is_promise (promise))
+  {
+    return jerry_throw (ecma_raise_type_error (ECMA_ERR_MSG (wrong_args_msg_p)));
+  }
+
+  ecma_value_t result = ecma_promise_get_result (ecma_get_object_from_value (promise));
```

Comment:
Cannot we return directly?

Reply:
Yes, I've updated the PR with the fixes.